### PR TITLE
KAFKA-8032: Wait for quota override removal in Quota tests 

### PR DESCRIPTION
**Pull request by omkreddy**
- Update UserQuotaTest/UserClientIdQuotaTest/ClientIdQuotaTest  tests

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
@rajinisivaram Does this look good to you?

**Comment by omkreddy** [comment]
still the test is failing. will check further.
```
java.lang.AssertionError: Client with id=QuotasTestProducer-1 should have been throttled
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at kafka.api.QuotaTestClients.verifyThrottleTimeMetric(BaseQuotaTest.scala:229)
	at kafka.api.QuotaTestClients.verifyProduceThrottle(BaseQuotaTest.scala:215)
	at kafka.api.BaseQuotaTest.testQuotaOverrideDelete(BaseQuotaTest.scala:124)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

File: core/src/test/scala/integration/kafka/api/ClientIdQuotaTest.scala [hunk 1]
```diff
@@ -59,6 +59,7 @@ class ClientIdQuotaTest extends BaseQuotaTest {
         val emptyProps = new Properties
         updateQuotaOverride(producerClientId, emptyProps)
         updateQuotaOverride(consumerClientId, emptyProps)
+        waitForQuotaUpdate(defaultProducerQuota, defaultConsumerQuota, Long.MaxValue)
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/ClientIdQuotaTest.scala:62]: Don't we need the same in `overrideQuotas`?

**Comment by ijuma** [comment]
@rajinisivaram it would be great for you to take a look at this small PR that should reduce test flakiness.

**Comment by rajinisivaram** [comment]
@omkreddy You said in the earlier note that the test was still failing. Do we know the reason for that?

**Comment by omkreddy** [comment]
@rajinisivaram I don't remember the reason :(. I am also, not sure, if this is the right fix.

**Pull request closed without merging**